ldappool/__init__.py

```diff
@@ -214,13 +214,13 @@ def _create_connector(self, bind, passwd):
                 conn.timeout = self.timeout
                 self._bind(conn, bind, passwd)
                 connected = True
-            except ldap.LDAPError, exc:
+            except ldap.LDAPError as exc:
                 time.sleep(self.retry_delay)
                 tries += 1
 
         if not connected:
-            if isinstance(exc, (ldap.NO_SUCH_OBJECT,
-                                ldap.INVALID_CREDENTIALS)):
+            if exc and isinstance(exc, (ldap.NO_SUCH_OBJECT,
```

Comment:
This doesn't work in Python3, exc will be deleted from locals when it goes out of scope after the exception handler:

```
>>> def func():
...   foo = None
...   try:
...     raise Exception()
...   except Exception as foo:
...     pass
...   return foo
... 
>>> func()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 7, in func
UnboundLocalError: local variable 'foo' referenced before assignment
```